module/os/windows/zfs/zfs_ioctl_os.c

```diff
@@ -129,8 +131,17 @@ NTSTATUS zpool_zfs_get_metrics(PDEVICE_OBJECT DeviceObject, PIRP Irp, PIO_STACK_
 	}
 
 	vdev_stat_t vs = { 0 };
+	unsigned __int64 mirror_slog_alloc_size = 0, mirror_slog_space = 0;
 	spa_config_enter(spa_perf, SCL_ALL, FTAG, RW_READER);
-	perf->mirror_slog_alloc_size = getMirrorSlogUsedSize(spa_perf->spa_root_vdev);
+	for (int i = 0; i < spa_perf->spa_l2cache.sav_count; i++) {
+	    if (spa_perf->spa_l2cache.sav_vdevs[i]->vdev_isl2cache)
+		perf->l2arc_space += spa_perf->spa_l2cache.sav_vdevs[i]->
+		vdev_stat.vs_space;
+	}
+	getMirrorSlogUsedSize(spa_perf->spa_root_vdev, &mirror_slog_alloc_size,
```

Comment:
Instead of passing a new local variable to get the size , we can pass the perf object variables (perf->mirror_slog_alloc_size , perf->mirror_slog_space )